### PR TITLE
Add steps for Shoot creation to docs/development/local_setup.md

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/area dev-productivity documentation
/kind task
/priority normal

**What this PR does / why we need it**:
Add steps for Shoot creation to `docs/development/local_setup.md`.

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: docs/usage/shoot_network_policies.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Network policies in the Shoot Cluster
 
-In addition to deploying network policies [into the Seed](https://github.com/gardener/gardener/tree/master/docs/development/network_policies.md),
+In addition to deploying network policies [into the Seed](https://github.com/gardener/gardener/tree/master/docs/development/seed_network_policies.md),
```
Comment (rfranzke) [inline docs/usage/shoot_network_policies.md]:
```suggestion
In addition to deploying network policies [into the Seed](../development/seed_network_policies.md),
```

File: docs/development/seed_network_policies.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Network Policies in the Seed Cluster
 
 This document describes the [Kubernetes network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) deployed by Gardener into the Seed cluster.
-For network policies deployed into the Shoot `kube-system` namespace, please see the [usage section](https://github.com/gardener/gardener/tree/master/docs/development/shoot_network_policies.md).
+For network policies deployed into the Shoot `kube-system` namespace, please see the [usage section](https://github.com/gardener/gardener/tree/master/docs/usage/shoot_network_policies.md).
```
Comment (rfranzke) [inline docs/development/seed_network_policies.md]:
```suggestion
For network policies deployed into the Shoot `kube-system` namespace, please see the [usage section](../usage/shoot_network_policies.md).
```

**Review by danielfoehrKn — DISMISSED** [review]

File: docs/development/local_setup.md [hunk 3]
```diff
@@ -310,34 +314,53 @@ to operate against your local running Gardener API Server.
...
+The `CloudProfile` resource is provider specific and describes the underlying cloud provider (available machine types, regions, machine images, etc.). Check the corresponding example manifest `dev/30-cloudprofile.yaml`. Check also the documentation and example manifests of the provider extension. Adapt `dev/30-cloudprofile.yaml` and apply it. 
+
+```bash
+kubectl apply -f dev/30-cloudprofile.yaml
+```
 
-## Configure Seed clusters for local development
+#### 4. Create the required ControllerRegistrations
+
+[Known Extension Implementations](../../README.md#infrastructure-provider) contains a list of available extension implementations. You need to create a ControllerRegistration for at least one infrastructure provider, dns provider (if the DNS for the Seed is not disabled), at least one operating system extension and at least one network plugin extension.
```
Comment (rfranzke) [inline docs/development/local_setup.md]:
```suggestion
The "[Known Extension Implementations](../../README.md#infrastructure-provider)" section contains a list of available extension implementations. You need to create a ControllerRegistration for at least one infrastructure provider, dns provider (if the DNS for the Seed is not disabled), at least one operating system extension and at least one network plugin extension.
```

dead link

File: docs/development/local_setup.md [hunk 4]
```diff
@@ -310,34 +314,53 @@ to operate against your local running Gardener API Server.
...
+As a convention, example ControllerRegistration manifest for an extension is located under `example/controller-registration.yaml` in the corresponding repository (for example for AWS the ControllerRegistration can be found [here](https://github.com/gardener/gardener-extension-provider-aws/blob/master/example/controller-registration.yaml)). An example creation of ControllerRegistration for provider-aws:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/gardener/gardener-extension-provider-aws/$VERSION/example/controller-registration.yaml
+```
+
+#### 5. Create a Seed
```
Comment (rfranzke) [inline docs/development/local_setup.md]: Should we call this "Register a Seed" to make more clear that Gardener does not create the seeds itself?

File: docs/development/local_setup.md [hunk 5]
```diff
@@ -347,32 +370,63 @@ the Gardenlet either needs to be supplied with a kubeconfig for the particular S
...
+
+Some time ago, we had a local setup based on VirtualBox/Vagrant.
+However, as we have progressed with the [Extensibility epic](https://github.com/gardener/gardener/issues/308) we noticed that this implementation/setup does no longer fit into how we envision external providers to be.
+Moreover, it hid too many things and came with a bunch of limitations, making the development scenario too "artificial":
+
+- No integration with machine-controller-manager.
+- The Shoot API Server is exposed via a NodePort. In a cloud setup a LoadBalancer would be used.
+- It was not possible to create Shoot clusters consisting of more than one worker node. Cluster auto-scaling therefore is not supported.
+- It was not possible to create two or more Shoot clusters in parallel.
+- The communication between the Seed and the Shoot Clusters uses VPN tunnel. In this setup tunnels are not needed since all components run on localhost.
```
Comment (rfranzke) [inline docs/development/local_setup.md]: Can this section be removed now?

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**